### PR TITLE
Refactor Com2IManagedPerPropertyBrowsingHandler to replace ArrayList with List<Atrribute>

### PR DESCRIPTION
**Pull request by Jericho**
Related: #8140

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8365)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
> I believe @JeremyKuhne is working on some Com2 stuff and wanted that namespace left alone for a bit.

That is true, both Com2 and AxHost related things should be avoided for now. I'll review this one, but please hold off on any cleanup in those types.

**Comment by Jericho** [comment]
Sorry about that. I wasn't aware.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs [hunk 1]
```diff
@@ -133,10 +132,10 @@ internal static unsafe Attribute[] GetComponentAttributes(VSSDK.IVSMDPerProperty
                         if (fi is not null && fi.IsStatic)
                         {
                             object? fieldValue = fi.GetValue(null);
-                            if (fieldValue is Attribute)
+                            if (fieldValue is Attribute fva)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs:135]: You should collapse these and please stay away from abbreviations. `if (fi.GetValue(null) is Attribute attribute)`

**Review by JeremyKuhne — CHANGES_REQUESTED** [review]
One minor comment

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs (hunk 1 above)
Comment (Jericho) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs]: Sure

File: src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs [hunk 2]
```diff
@@ -187,7 +185,7 @@ internal static unsafe Attribute[] GetComponentAttributes(VSSDK.IVSMDPerProperty
                     try
                     {
                         attr = (Attribute?)Activator.CreateInstance(t);
-                        attrs.Add(attr);
+                        if (attr is not null) attrs.Add(attr);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs:188]: Line 198 to 200 - > attrs.ToArray()

Comment (Jericho) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComponentModel/COM2Interop/COM2IManagedPerPropertyBrowsingHandler.cs:188]: Done.

**Review by dreddy-work — APPROVED** [review]

**Review by JeremyKuhne — APPROVED** [review]
:shipit:

**Pull request merged**